presto-main/src/main/java/com/facebook/presto/server/ServerMainModule.java

```diff
@@ -209,6 +212,29 @@ protected void setup(Binder binder)
             config.setKeyStorePassword(internalCommunicationConfig.getKeyStorePassword());
         });
 
+        if (internalCommunicationConfig.isKerberosEnabled()) {
+            File kerberosConfig = internalCommunicationConfig.getKerberosConfig();
+            File kerberosKeytab = internalCommunicationConfig.getKerberosKeytab();
+            String kerberosPrincipal = internalCommunicationConfig.getKerberosPrincipal();
+            String kerberosServiceName = internalCommunicationConfig.getKerberosServiceName();
+            checkArgument(kerberosConfig != null, "kerberos config must be set");
```

Comment:
Add these as `@NotNull` validation constraints to the config class. This will provide appropriate error messages when the values are not set. Then you can remove the checks here since we know the config system will only return a validated bean.

Reply:
Ah, I see that this validation needs to be conditional. You can put these new config properties (except for the enabled flag) in a separate class (`KerberosInternalCommunicationConfig` or `KerberosCommunicationConfig`?) that only gets created if Kerberos is enabled.

We do a similar trick in `HiveAuthenticationModule` where the various modules with their configs get bound conditionally if they are enabled.